src/pfe/file-watcher/server/src/controllers/projectEventsController.ts

```diff
@@ -213,16 +216,20 @@ export async function updateProjectForNewChange(projectID: string, timestamp: nu
 
             if (newChunkRemaining == 0) {
                 // this is the last chunk for this timestamp, check if still waiting for other chunks for other timestamps
+                logger.logProjectTrace("Received last chunk for timestamp " + timestamp + " , checking if still waiting for other chunks for other timestamps... ", projectID);
                 let shouldTriggerBuild = true;
                 if (chunkRemainingMap.size != 0) {
                     const chunkRemainingArray = chunkRemainingMap.get(projectID);
                     if (chunkRemainingArray && chunkRemainingArray.length > 0) {
                         // still waiting for some chunks
+                        logger.logProjectTrace("Still waiting for other chunks... ", projectID);
                         shouldTriggerBuild = false;
                     }
                 }
                 if (shouldTriggerBuild) {
+                    logger.logProjectInfo("Received all chunks", projectID);
                     if (projectInfo.autoBuildEnabled) {
+                        logger.logProjectInfo("Proceeding the build... ", projectID);
                         if (!statusController.isBuildInProgressOrQueued(projectID)) {
                             const operation = new projectOperation.Operation("update", projectInfo);
```

Comment:
Can you add a log message at the end of this if block as well? Would give us another timestamp to see where time is being spent during this whole process.